trees/src/iavl/tree.rs

```diff
@@ -179,6 +179,40 @@ impl Node {
         }
     }
 
+    /// Returns the leftmost leaf key in the subtree rooted at this node AKA the smallest key in the subtree.
+    fn leftmost_leaf_key<T: Database>(&self, node_db: &NodeDB<T>) -> Vec<u8> {
+        match self {
+            Node::Leaf(leaf) => return leaf.key.clone(),
+            Node::Inner(inner) => {
+                let mut left_node = match &inner.left_node {
+                    Some(inner_left_node) => inner_left_node,
+                    None => &node_db
+                        .get_node(&inner.left_hash)
+                        .expect("node db should contain all nodes"),
+                };
+
+                let mut cached_node;
```

Comment:
You may use `break` to return value from `loop`

Reply:
Changed.